core/src/main/java/nl/nn/adapterframework/jta/narayana/PoolingDataSourceJDBCAccess.java

```diff
@@ -0,0 +1,91 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.jta.narayana;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.StringTokenizer;
+
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+
+import org.apache.commons.dbcp2.ConnectionFactory;
+import org.apache.commons.dbcp2.DataSourceConnectionFactory;
+import org.apache.commons.dbcp2.PoolableConnection;
+import org.apache.commons.dbcp2.PoolableConnectionFactory;
+import org.apache.commons.dbcp2.PoolingDataSource;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+
+import com.arjuna.ats.arjuna.exceptions.FatalError;
+import com.arjuna.ats.arjuna.objectstore.jdbc.JDBCAccess;
+import com.arjuna.ats.internal.arjuna.objectstore.jdbc.accessors.DataSourceJDBCAccess;
+
+import nl.nn.adapterframework.jndi.ObjectLocator;
+
+/**
+ * Alternative to {@link DataSourceJDBCAccess} which does a JNDI lookup each time a connection is called and does not pool connections.
```

Comment:
```suggestion
 * Alternative to {@link DataSourceJDBCAccess} that adds connection pooling, instead of doing a JNDI lookup each time a connection is called.
```